lib/Twig/Extension/Core.php

```diff
@@ -611,6 +612,33 @@ function twig_array_merge($arr1, $arr2)
     return array_merge($arr1, $arr2);
 }
 
+/**
+ * Return the values from a single column in the input array.
+ *
+ * <pre>
+ *  {% set items = [{ 'fruit' : 'apple'}, {'fruit' : 'orange' }] %}
+ *
+ *  {% set fruits = items|column('fruit') %}
+ *
+ *  {# fruits now contains ['apple', 'orange'] #}
+ * </pre>
+ *
+ * @param array|Traversable $arr        An array
+ * @param mixed             $column_key The column key
+ *
+ * @return array The array of values
+ */
+function twig_array_column($arr, $column_key)
+{
+    if ($arr instanceof Traversable) {
+        $arr = iterator_to_array($arr);
```

Comment:
This will consume the iterator if it is not rewindable, right? Is that something the user will actually expect?

Reply:
Sine the purpose of the filter is to get some data from the iterator, I think consuming it is the expected behavior.

Reply:
I added this to bring the feature in line with the [random](https://github.com/twigphp/Twig/blob/60a8f465d6f259c92a8fe0fb61cf8f767bd8d025/lib/Twig/Extension/Core.php#L299), [replace](https://github.com/twigphp/Twig/blob/60a8f465d6f259c92a8fe0fb61cf8f767bd8d025/lib/Twig/Extension/Core.php#L450), [merge](https://github.com/twigphp/Twig/blob/60a8f465d6f259c92a8fe0fb61cf8f767bd8d025/lib/Twig/Extension/Core.php#L548-L554), [slice](https://github.com/twigphp/Twig/blob/60a8f465d6f259c92a8fe0fb61cf8f767bd8d025/lib/Twig/Extension/Core.php#L582-L588), [join](https://github.com/twigphp/Twig/blob/60a8f465d6f259c92a8fe0fb61cf8f767bd8d025/lib/Twig/Extension/Core.php#L651), [reverse](https://github.com/twigphp/Twig/blob/60a8f465d6f259c92a8fe0fb61cf8f767bd8d025/lib/Twig/Extension/Core.php#L779), [sort](https://github.com/twigphp/Twig/blob/60a8f465d6f259c92a8fe0fb61cf8f767bd8d025/lib/Twig/Extension/Core.php#L815), and [batch](https://github.com/twigphp/Twig/blob/60a8f465d6f259c92a8fe0fb61cf8f767bd8d025/lib/Twig/Extension/Core.php#L1417) functions/filters.

Reply:
@xabbuh the behaviour is the same of [random](https://github.com/twigphp/Twig/blob/60a8f465d6f259c92a8fe0fb61cf8f767bd8d025/lib/Twig/Extension/Core.php#L299)